hs_core/views/__init__.py

```diff
@@ -475,18 +475,57 @@ def my_resources(request, page):
         edit_permission = frm.cleaned_data['edit_permission'] or False
         published = frm.cleaned_data['published'] or False
         startno = frm.cleaned_data['start']
-        if(startno < 0):
-            startno = 0
-        start = startno or 0
-        from_date = frm.cleaned_data['from_date'] or None
-        words = request.REQUEST.get('text', None)
-        public = not request.user.is_authenticated()
 
         search_items = dict(
             (item_type, [t.strip() for t in request.REQUEST.getlist(item_type)])
             for item_type in ("type", "author", "contributor", "subject")
         )
 
+        from_date = frm.cleaned_data['from_date'] or None
+        words = request.REQUEST.get('text', None)
+
+        if startno is None:
+            # this is the case where either the 'RESOURCES' menu option or one of the 3 filtering menu options
+            # (Owned by me, Editable by me, Viewable by me) has been selected, OR a search item has been entered
+            startno = 0
+            if owner:   # 'Owned by me' selected
+                request.session['owner'] = owner
+                request.session['edit_permission'] = edit_permission
+            else:   # either 'Editable by me' or 'Viewable by me' selected
+                if 'owner' in request.session:
+                    del request.session['owner']
+                if 'edit_permission' in request.session:
+                    del request.session['edit_permission']
+
+            if not owner and edit_permission:   # 'Editable by me' selected
+                request.session['edit_permission'] = edit_permission
```

Comment:
In fact, I think you can preserve 'edit_permission' session variable without deleting it since it is used not only by 'Owned by me' case, but also used by 'Editable by me' case, right?